### PR TITLE
feat(todoist): close cadence task on non-Todoist outreach (#268 follow-up)

### PR DESCRIPTION
**Pull request by spengrah**
## Summary

- Add outreach detection to the Todoist reconciler so cadence tasks (outreach reminders) are automatically closed when the user reaches out via a non-Todoist source (e.g., Telegram)
- Introduce `wasReachedOutSinceSync` function and `synced_last_outreach_at` metadata key, mirroring the existing `wasContactedSinceSync` / `synced_last_contacted` pattern
- Restructure `reconcileContactTasks` to check outreach **before** the follow-up gate, so cadence tasks are closed even when a pending follow-up exists
- Extract `closeOnOutreach` helper with `([]SyncCommand, bool)` return to safely distinguish "no outreach" from "outreach handled but no Todoist command needed" (pending temp ID case)

**Plan:** `.ai/log/plan/cadence-task-close-on-outreach.md`

## Test plan

- [x] 7 new integration tests in `provider_outreach_close_test.go`:
  - Close cadence task with pending follow-up (primary regression)
  - No close when no new outreach (negative case)
  - Close without follow-up + recreation on next tick
  - Legacy task backfill without closing (migration test)
  - State failure skips close (safety test)
  - Pending temp ID: handled=true, no item_close, DB transition
  - Nil last_outreach_at edge case
- [x] All 40 todoist package tests pass (33 existing + 7 new)
- [x] `make lint` clean
- [x] `make test` all pass
- [x] `make test-e2e-diff` all pass (20 E2E tests)
- [x] Codex review: PASS after 3 rounds (fixed nil-return ambiguity bug)
- [x] Manual verification after deploy: send Telegram message to contact with cadence task, verify task closes on next sync tick

### CODE REVIEW

File: backend/internal/todoist/provider.go [hunk 1]
```diff
@@ -928,6 +951,75 @@ func (p *CadenceSyncProvider) reconcileContactTasks(
...
+			Str("contactName", contact.FullName).
+			Msg("contact was reached out to since last sync (non-Todoist source), closing cadence task")
+
+		// Mark completed locally FIRST — only enqueue the Todoist close if this succeeds.
+		// If state update fails, skip the remote close entirely to avoid leaving a
+		// local 'managed' row while the Todoist task is closed (same pattern as
+		// handleFollowUpDismissal).
+		if _, err := p.contactTaskRepo.UpdateContactTaskState(ctx, task.ID, repository.ContactTaskStateCompleted); err != nil {
+			logger.Warn().Err(err).Str("contactId", contact.ID.String()).Msg("failed to mark cadence task completed after outreach detection — skipping close")
+			return nil, false
```
[the remark this answers is not in the document]
Comment (spengrah) [reply backend/internal/todoist/provider.go:977]: Not a new risk. `tryRecoverPendingTempID` runs in `processItems` (line 421), which executes **before** `reconcileContactTasks` in the sync loop (provider.go:209-250). So recovery always has a chance to resolve the real ID before `closeOnOutreach` runs.

The only scenario where this could matter is if the Todoist item hasn't appeared in the sync response yet (API hasn't returned it). In that case, `tryRecoverPendingTempID` can't fire regardless — the item isn't in the response. And `closeOnOutreach` with a pending temp ID correctly returns `(nil, true)`: marks the local row completed (so the reconciler doesn't try to recreate it), but skips `item_close` (no real ID). On the next tick when the item does appear, `processItem` sees no matching local row and ignores it — the same orphan handling that exists today for any task completed between sync cycles.

This exposure is identical to `handleTaskCompletion` which also marks cadence tasks completed immediately.
